src/libraries/PCPricing.sol

```diff
@@ -0,0 +1,303 @@
+    /// @dev If trackedCurrencies length exceeds batchSize, it will use batch processing
+    function adjustAllNonAnchorPrices(
+        PricingData storage self,
+        bool increase
+    ) internal {
+        uint256 length = self.trackedCurrencies.length;
+
+        // If batchSize is not set or array is smaller than batch size, process all at once
+        if (self.batchSize == 0 || length <= self.batchSize) {
+            for (uint256 i = 0; i < length; i++) {
+                bytes memory currency = self.trackedCurrencies[i];
+                if (keccak256(currency) != keccak256(self.anchorCurrency)) {
+                    adjustCurrencyPrice(self, currency, increase);
+                }
+            }
+            emit NonAnchorPricesReduced();
+        } else {
+            adjustNonAnchorPricesBatch(self, increase, self.batchSize);
+        }
+    }
+
+    /// @notice Get the current batch processing state
+    /// @return lastProcessedIndex The last processed index
+    /// @return totalCurrencies The total number of currencies
+    function getBatchProcessingState(
+        PricingData storage self
+    )
+        internal
+        view
+        returns (uint256 lastProcessedIndex, uint256 totalCurrencies)
+    {
+        return (self.lastProcessedIndex, self.trackedCurrencies.length);
+    }
+
+    /// @notice Get the current price of a currency
+    /// @param currency The currency to get the price of
+    /// @return price The price of the currency
+    function getCurrencyPrice(
+        PricingData storage self,
+        bytes memory currency
+    ) internal view returns (uint256 price) {
+        require(self.currencyPrice[currency] > 0, "Currency price not set");
+        return self.currencyPrice[currency];
+    }
+
+    /// @notice Get all tracked currency prices
+    /// @return currencies The list of tracked currencies
+    /// @return prices The list of prices for the tracked currencies
+    function getAllCurrencyPrices(
+        PricingData storage self
+    ) internal view returns (bytes[] memory, uint256[] memory) {
+        uint256 length = self.trackedCurrencies.length;
+        bytes[] memory currencies = new bytes[](length);
+        uint256[] memory prices = new uint256[](length);
+
+        for (uint256 i = 0; i < length; i++) {
+            bytes memory currency = self.trackedCurrencies[i];
+            currencies[i] = currency;
+            prices[i] = self.currencyPrice[currency];
+        }
+
+        return (currencies, prices);
+    }
+
+    /// @notice Check if a currency exists
+    /// @param currency The currency to check
+    /// @return exists Whether the currency exists
+    function currencyExists(
+        PricingData storage self,
+        bytes memory currency
+    ) internal view returns (bool) {
+        return self.currencyPrice[currency] > 0;
+    }
+
+    /// @notice Remove a currency from the pricing data
+    /// @param currency The currency to remove
+    function removeCurrency(
```

Comment:
What happens if you remove a currency in the middle of a batch of adjustments? Is there a test for this?